packages/libs/error-stack/src/fmt/hook.rs

```diff
@@ -175,26 +249,271 @@ impl HookContextImpl {
+    ///     code: usize,
+    ///     reason: &'static str,
+    /// }
+    ///
+    /// Report::install_debug_hook::<Error>(|Error { code, reason }, ctx| {
+    ///     if ctx.alternate() {
+    ///         ctx.snippet(format!("Error {code}:\n  {reason}"));
+    ///     }
+    ///
+    ///     ctx.emit(format!("Error {code}"));
+    /// });
+    ///
+    /// let report = Report::new(std::io::Error::from(ErrorKind::InvalidInput))
+    ///     .attach(Error {
+    ///         code: 404,
+    ///         reason: "Not Found - Server cannot find requested resource",
+    ///     })
+    ///     .attach(Error {
+    ///         code: 405,
+    ///         reason: "Bad Request - Server cannot or will not process request",
+    ///     });
+    ///
+    /// # owo_colors::set_override(true);
+    /// # fn render(value: String) -> String {
+    /// #     let backtrace = regex::Regex::new(r"Backtrace No\. (\d+)\n(?:  .*\n)*  .*").unwrap();
+    /// #     let backtrace_info = regex::Regex::new(r"backtrace with (\d+) frames \((\d+)\)").unwrap();
+    /// #
+    /// #     let value = backtrace.replace_all(&value, "Backtrace No. $1\n  [redacted]");
+    /// #     let value = backtrace_info.replace_all(value.as_ref(), "backtrace with [n] frames ($2)");
+    /// #
+    /// #     ansi_to_html::convert_escaped(value.as_ref()).unwrap()
+    /// # }
+    /// #
+    /// # #[cfg(rust_1_65)]
+    /// # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__hookcontext_emit.snap")].assert_eq(&render(format!("{report:#?}")));
+    /// #
+    /// println!("{report:#?}");
+    /// ```
+    ///
+    /// <pre>
+    #[doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__hookcontext_emit.snap"))]
+    /// </pre>
+    pub fn snippet(&mut self, snippet: impl Into<String>) {
+        self.inner.snippets.push(Emit::immediate(snippet));
+    }
+
+    /// Create a new multiline snippet, which is going to be appended to the end of the message, but
+    /// deferred, until the end of the stack of current snippets.
+    ///
+    /// A stack are all attachments until a [`Context`] is encountered in the frame stack,
+    /// lines added via this function are going to be emitted at the end.
+    ///
+    /// [`Context`]: crate::Context
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # // we only test with Rust 1.65, which means that `render()` is unused on earlier version
+    /// # #![cfg_attr(not(rust_1_65), allow(dead_code, unused_variables, unused_imports))]
+    /// use std::io::ErrorKind;
+    ///
+    /// use error_stack::Report;
+    ///
+    /// struct Error {
+    ///     code: usize,
+    ///     reason: &'static str,
+    /// }
+    ///
+    /// struct Suggestion(&'static str);
+    ///
+    /// Report::install_debug_hook::<Error>(|Error { code, reason }, ctx| {
+    ///     if ctx.alternate() {
+    ///         ctx.snippet_deferred(format!("Error {code}:\n  {reason}"));
+    ///     }
```

Comment:
```suggestion
    ///     if ctx.alternate() {
    ///         // Add a snippet, which is printed after other snippets
    ///         ctx.snippet_deferred(format!("Error {code}:\n  {reason}"));
    ///     }
```

Reply:
Would be great if you could look for other places as well, I think especially the `cast()` option is a good place to add some explanations as well.